### DISCUSSION TITLE
Push protection for existing secrets

### DISCUSSION THREAD

**Discussion by john-yacuta-submittable** [community/community#72670] [Code Security] [answered]
### Select Topic Area

Question

### Body

I understand that push protection for repositories prevents secrets from being pushed to it. However, does this push protection feature also include this for existing secrets? 

Example scenario:
1. A developer who contributes to an existing repository contains a secret flagged by Secrets Scanning but is unaware it exists. 
2. The organization enables push protection for the whole organization outlined in the steps [here](https://docs.github.com/en/enterprise-cloud@latest/code-security/secret-scanning/push-protection-for-repositories-and-organizations#enabling-secret-scanning-as-a-push-protection-for-an-organization).
3. Question: Is the developer blocked from pushing his/her changes to the repository I mentioned above? Notice that they didnt push the secret but because it exists already in the repository they contribute to it prevents them from pushing other code changes since its flagged. This would mean they would have to resolve the secret and then they can push their code changes. 

I ask this because it might be disruptive to contributors. Thanks!

**Comment by aashah** [comment, marked as answer]
If a secret scanning alert already exists, then push protection should not block the developer's push. Only if the push would contain a new alert will it block.
